### DISCUSSION TITLE
Pokemon Violet/Scarlet corrupted player photo

### DISCUSSION THREAD

**Discussion by pasqualenardiello** [kwsch/PKHeX#3858] [Q&A] [answered]
Hi everyone, today a very strange thing happened after I tried to change the Photo blocks in S/V.
Long story short: my Scarlet save file was lost after I changed micro-SD, and the backup I had was not very recent. So I took a post-game save file I had, which happened to be a Violet one, changed game version, TID, SID etc... and everything worked. After some playtime I remembered the enrollmentDate and the trainer photo and icon were still the same from violet, so with  PkHex I correctly changed the enrollment date and reimported the photo and icon blocks from the old Scarlet backup. I even did some digging and found the block address of the original trainer icon, as reported in the first line of the screenshot:

![photo_compSV](https://user-images.githubusercontent.com/6310546/232232872-5e367a86-9d75-4574-9f69-445817463dc5.png)

Now the problem is, even if the original trainer photo and icon actually changed, they look corrupted:

![2023041423153500-1A552196CBAEC7F073BF13D42EA6C787](https://user-images.githubusercontent.com/6310546/232233023-840a460a-9d7b-4dd7-9bf5-031f2e01c5a2.jpg)

Has anyone any idea why this happens?(Sorry for the long post btw, and thank you for the amazing work!)

**Comment by sora10pls** [comment, marked as answer]
There are several other save blocks involved with initial and current trainer icons/photos, I'll label them in a bit.

Edit: https://github.com/kwsch/PKHeX/commit/f433d020529c0a357a024450a12fb486361848b5 should address this, copy all of the save blocks from your old save file to the new one.

**Reply by pasqualenardiello** [reply]
Thank you, I'll try as soon as I get home!

Edit: changed all the blocks listed in the commit and worked like a charm. Awesome!
